build-ps/debian/extra/mysql-systemd

```diff
@@ -0,0 +1,374 @@
+                else
+                    log_warning_msg "Another instance of mysqld running on $mysql_pid, exiting.."
+                fi
+                exit 1
+
+	    fi
+    fi
+}
+
+# Run mysqld with --wsrep-recover and parse recovered position from log.
+# Position will be stored in wsrep_start_position_opt global.
+wsrep_start_position_opt=""
+wsrep_recover_position() {
+  local mysqld_cmd="/usr/sbin/mysqld"
+  local ret=0
+  local uuid=""
+  local seqno=0
+  local wr_logfile=$(mktemp $datadir/wsrep_recovery.XXXXXX)
+  local err_log=$(parse_cnf log-error server mysqld)
+  if [ -f $grastate_loc ]; then
+    uuid=$(grep 'uuid:' $grastate_loc | cut -d: -f2 | tr -d ' ')
+    seqno=$(grep 'seqno:' $grastate_loc | cut -d: -f2 | tr -d ' ')
+
+  # If sequence number is not equal to -1, wsrep-recover co-ordinates aren't used.
+  # So, directly pass whatever is obtained from grastate.dat
+    if [ ! -z $seqno ] && [ $seqno -ne -1 ]; then
+      log_info_msg "Skipping wsrep-recover for $uuid:$seqno pair" >> $wr_logfile
+      log_info_msg "Assigning $uuid:$seqno to wsrep_start_position" >> $wr_logfile
+
+      wsrep_start_position_opt="--wsrep_start_position=$uuid:$seqno"
+      if [ -f $err_log ]; then
+        cat $wr_logfile >> $err_log
+      fi
+      rm $wr_logfile
+
+      echo ${wsrep_start_position_opt}
+      return $ret
+    fi
+  fi
+  local euid=$(id -u)
+
+  [ "$euid" = "0" ] && chown 'mysql' $wr_logfile
+  chmod 600 $wr_logfile
+
+  local wr_pidfile="$datadir/"`hostname`"-recover.pid"
+
+  local wr_options="--log_error='$wr_logfile' --pid-file='$wr_pidfile'"
+
+  log_info_msg "WSREP: Running position recovery with $wr_options" >> $wr_logfile
+
+  eval "$mysqld_cmd --user=mysql --log-error-verbosity=3 --wsrep_recover $wr_options"
+  local rp="$(grep '\[WSREP\] Recovered position:' $wr_logfile)"
+  if [ -z "$rp" ]; then
+    local skipped="$(grep WSREP $wr_logfile | grep 'skipping position recovery')"
+    if [ -z "$skipped" ]; then
+      log_failure_msg "WSREP: Failed to recover position: " >> $wr_logfile
+      ret=2
+    else
+      log_info_msg "WSREP: Position recovery skipped" >> $wr_logfile
+    fi
+  else
+    local start_pos="$(echo $rp | sed 's/.*WSREP\]\ Recovered\ position://' \
+        | sed 's/^[ \t]*//')"
+    log_info_msg "WSREP: Recovered position $start_pos" >> $wr_logfile
+    wsrep_start_position_opt="--wsrep_start_position=$start_pos"
+  fi
+
+  if [ -f $err_log ]; then
+      echo "Log of wsrep recovery (--wsrep-recover):" >> $err_log
+      cat $wr_logfile >> $err_log
+  fi
+  rm $wr_logfile
+
+  echo ${wsrep_start_position_opt}
+  return $ret
+}
+
+
+ret=0
+
+case $action in
+    "galera-recovery") wsrep_recover_position ;;
+    "start-pre") check_running; install_db ;;
+    "start-post")
+      wait_for_pid created  "$pid_path"; ret=$?
+      if [[ $ret -eq 1 ]];then
```

Comment:
also please add message for successful start

Reply:
We do have messages about successful start, they look as follows:
Dec 17 14:00:41 pxc80-alex-3.ci.percona.com systemd[1]: Starting Percona XtraDB Cluster...
Dec 17 14:00:48 pxc80-alex-3.ci.percona.com mysql-systemd[4156]: SUCCESS!
Dec 17 14:00:48 pxc80-alex-3.ci.percona.com systemd[1]: Started Percona XtraDB Cluster.